### PR TITLE
Updated to 10.3.18

### PR DESCRIPTION
**Pull request by RahulRavishankar**

### CODE REVIEW

File: mariadb103.spec [hunk 1]
```diff
@@ -116,7 +116,7 @@
 %bcond_without unbundled_pcre
 %else
 %bcond_with unbundled_pcre
-%global pcre_bundled_version 8.43
+%global pcre_bundled_version 10.33
```
Comment (carlwgeorge) [inline mariadb103.spec]: The bundled PCRE version is defined upstream in [pcre/configure.ac](https://github.com/MariaDB/server/blob/mariadb-10.3.18/pcre/configure.ac#L11-L12).  It is still 8.43 at the mariadb-10.3.18 tag, so we can keep this line the same.

File: mariadb103.spec [hunk 2]
```diff
@@ -1626,6 +1626,9 @@ fi
 %endif
 
 %changelog
+*Tue Oct 01 2019 Rahul <rahul@rahul-computer> - 3:10.3.18-1
```
Comment (carlwgeorge) [inline mariadb103.spec]: Add a space between the asterisk and Tue.

**Review by carlwgeorge — CHANGES_REQUESTED** [review]
With the minor adjustments I commented on, your changes look good overall.  Unfortunately the build doesn't pass.

https://cirrus-ci.com/task/4885201999626240

Patch13 doesn't apply cleanly.  How do you feel about rebasing the patch on the 10.3.18 code base?  If you haven't done that before I can make that change and push it to your pull request branch to be included with this change before merging.

**Comment by RahulRavishankar** [comment]
> With the minor adjustments I commented on, your changes look good overall. Unfortunately the build doesn't pass.
> 
> https://cirrus-ci.com/task/4885201999626240
> 
> Patch13 doesn't apply cleanly. How do you feel about rebasing the patch on the 10.3.18 code base? If you haven't done that before I can make that change and push it to your pull request branch to be included with this change before merging.

Can you help me out with rebasing Patch 13? 
Can you share a few links to help me get started?
I'm a newbie!

**Comment by carlwgeorge** [comment]
Rebasing patches is one of the trickier tasks with packaging, especially with something as large as MariaDB.  Unfortunately that makes this a terrible task for a newbie.  I'm not aware of specific guidelines or walkthroughs for rebasing patches in the context of RPM packaging.  These links may be a good starting point for patches in general.

- https://linuxacademy.com/blog/linux/introduction-using-diff-and-patch/
- https://git-scm.com/docs/git-format-patch
- https://www.oreilly.com/library/view/git-pocket-guide/9781449327507/ch11.html
- https://en.wikipedia.org/wiki/Patch_(Unix)

Fedora is going to have to rebase the patch as well to update to 10.3.18, so I did the work for that and sent them a [pull request](https://src.fedoraproject.org/rpms/mariadb/pull-request/15).  Spoiler alert, the patch file I rebased for Fedora is [here](https://src.fedoraproject.org/fork/carlwgeorge/rpms/mariadb/blob/update_10.3.18/f/mariadb-spider_on_armv7hl.patch) and we can reuse it in this pull request.  Here are the basic steps I took.

1. I opened the [existing patch](https://github.com/iusrepo/mariadb103/blob/10.3.17-1/mariadb-spider_on_armv7hl.patch) and saw that it was created with the `git format-patch` command.  You can tell that because of the email header and the lines that start with `diff --git`.  The patch file also has a note about an [upstream commit](https://github.com/MariaDB/server/commit/80f9bd3d9c4f1b6367487a88e4ca816fcb8ce186) it was based on.  That upstream commit isn't in a branch, but by looking at the file changes and using `git blame`, I found that it was in the 10.5 branch with a [different hash](https://github.com/MariaDB/server/commit/ddce85907611e0533d6226de7f53e751cf173f6a).  It was likely rebased on the tip of 10.5 when it was merged, which changed the parent and thus change the commit reference.

2. I cloned the [MariaDB/server](https://github.com/MariaDB/server) repo, checked out the mariadb-10.3.18 tag, created a local branch from that, and cherry picked the previous commit on top of it.  It applied cleanly, but if it hadn't I would have had to resolve any merge conflicts.

3. I created a new patch file with the command `git format-patch -1 HEAD`, and then added the extra notes that the existing patch file had to the top of it.

I went ahead and amended your commit to include the rebased patch.  Thanks for taking a stab at this, I hope this explanation is helpful.

**Review by carlwgeorge — APPROVED** [review]

**Comment by carlwgeorge** [comment]
mariadb103-10.3.18-1.el7.ius has been published to the testing repo.

**Pull request merged**